src/Servers/Kestrel/Core/src/Internal/Http3/Http3ControlStream.cs

```diff
@@ -283,7 +278,7 @@ private ValueTask ProcessSettingsFrameAsync(ReadOnlySequence<byte> payload)
             }
 
             _haveReceivedSettingsFrame = true;
-            using var closedRegistration = _context.ConnectionContext.ConnectionClosed.Register(state => ((Http3ControlStream)state).OnStreamClosed(), this);
+            using var closedRegistration = _context.StreamContext.ConnectionClosed.Register(state => ((Http3ControlStream)state).OnStreamClosed(), this);
```

Comment:
What's thee difference between OnStreamClosed() and OnInputOrOutputCompleted() supposed to be? I see that OnStreamClosed() calls Abort(ConnectionAbortedException ex) which currently no-ops.